src/main/python/rlbot/agents/hivemind/python_hivemind.py

```diff
@@ -0,0 +1,157 @@
+
+        # Give a warning if exec_path was given.
+        if 'exec_path' in options:
+            self.logger.warning(
+                """An exec_path was given, but this is a PythonHivemind, and so it is ignored.
+                Try subclassing SubprocessHivemind if you want to use an executable."""
+            )
+
+        # The game interface is how you get access to things
+        # like ball prediction, the game tick packet, or rendering.
+        self.game_interface = GameInterface(self.logger)
+
+        # drone_indices is a set of bot indices
+        # which requested this hivemind with the same key.
+        self.drone_indices = set()
+
+    def try_receive_agent_metadata(self):
+        """Adds all drones with the correct key to our set of running indices."""
+        while not self.metadata_queue.empty():
+            # Tries to get the next agent from the queue.
+            single_agent_metadata: AgentMetadata = self.metadata_queue.get(timeout=1.0)
+            # Adds the index to the drone_indices.
+            self.drone_indices.add(single_agent_metadata.index)
+
+    def start(self):
+        """Starts the BotHelperProcess - Hivemind."""
+        # Prints an activation message into the console.
+        # This lets you know that the process is up and running.
+        self.logger.debug("Hello, world!")
+
+        # Loads game interface.
+        self.game_interface.load_interface()
+
+        # Collect drone indices that requested a helper process with our key.
+        self.logger.info("Collecting drones; give me a moment.")
+        self.try_receive_agent_metadata()
+        self.logger.info("Ready to go!")
+
+        # Runs the game loop where the hivemind will spend the rest of its time.
+        self.__game_loop()
+
+    def __game_loop(self):
+        """
+        The bot hivemind will stay in this loop for the whole game. 
+        This is where the initialize_hive and get_outputs functions are called.
+        """
+
+        # Creating ball prediction and field info objects to later update in wrapper methods.
+        self._ball_prediction = BallPrediction()
+        self._field_info = FieldInfoPacket()
+        self.game_interface.update_field_info_packet(self._field_info)
+        # Wrapper for renderer.
+        self.renderer: RenderingManager = self.game_interface.renderer
+
+        # Create packet object.
+        packet = GameTickPacket()
+        self.game_interface.update_live_data_packet(packet)
+
+        # Initialization step for your hivemind.
+        self.initialize_hive(packet)
+
+        while True:
+            previous_packet_time = packet.game_info.seconds_elapsed
+
+            # Updating the packet.
+            self.game_interface.update_live_data_packet(packet)
+
+            # Checking if packet is old. If yes, sleep and skip running the code.
+            if previous_packet_time == packet.game_info.seconds_elapsed:
+                time.sleep(0.003)
+                continue
```

Comment:
If you use fresh_live_data_packet here too, you won't need the time checking code since that all happens internally.

Reply:
That simplifies things nicely!